src/writer/xapianHandler.cpp

```diff
@@ -95,22 +95,34 @@ std::unique_ptr<ContentProvider> TitleXapianHandler::getContentProvider() const
 
 void TitleXapianHandler::handle(Dirent* dirent, const Hints& hints)
 {
-  handle_dirent(dirent);
-}
+  // We have no items to get the title from. So it is a redirect
+  // We assume that if the redirect has a title, we must index it.
+  if (dirent->getNamespace() != 'C') {
+    // We should always have namespace == 'C' but let's be careful.
+    return;
+  }
 
-void TitleXapianHandler::handle(Dirent* dirent, std::shared_ptr<Item> item)
-{
-  handle_dirent(dirent);
+  auto title = dirent->getRealTitle();
+  if (title.empty()) {
+    return;
+  }
+  auto path = dirent->getPath();
+  mp_indexer->indexTitle(path, title);
 }
 
-void TitleXapianHandler::handle_dirent(Dirent* dirent)
+void TitleXapianHandler::handle(Dirent* dirent, std::shared_ptr<Item> item)
 {
+  // We have a item. And items have indexData. We must use it.
   if (dirent->getNamespace() != 'C') {
     // We should always have namespace == 'C' but let's be careful.
     return;
   }
 
-  auto title = dirent->getRealTitle();
+  auto indexData = item->getIndexData();
+  if (!indexData || !indexData->hasIndexData()) {
+    return;
+  }
+  auto title = indexData->getTitle();
```

Comment:
1. For an item with indexData, can a title obtained in this way be different from the title stored in the dirent?
2. Don't items without indexData deserve being represented in the title index in any case?

Reply:
> For an item with indexData, can a title obtained in this way be different from the title stored in the dirent?

Yes, actually, this already differ. The title from indexData is without accent.

> Don't items without indexData deserve being represented in the title index in any case?

No. With zim file created from warc file, we don't want to index the article but (html) redirection to them. So we will have item with data and title we don't want to index in any way.

But we should ask if the coherence should be put on the FRONT_ARTICLE hints.
And the answer is probably more functional that technique. @kelson42 and @rgaudin may have they through on this :

A front article (put in `listing/titleOrdered/v1`) must probably also being put in the title xapian index.

- Should we titleindex an article with a (indexable) title but not FRONT_ARTICLE ?
- Should we titleindex a FRONT_ARTICLE article without indexdata (or just without indexable title) (using its dirent->title) ?

For the first one, I would say yes but I not sure.
For the second I would say it is a incoherent request and is a bug in the data coming from the user (either a bug in user implemented IndexData or incoherence between item's title and FRONT_ARTICLE)

And next question : Do we want to enforce this coherence at libzim level or scrapper level ?
(A scrapper may want to have all its items FRONT_ARTICLE but titleindex only few of them. It would be surprising, but not to us to discard this feature)